src/android/DeviceAccounts.java

```diff
@@ -92,9 +97,20 @@ private JSONArray formatResult(List<Account> accounts) throws JSONException {
     return Accounts;
   }
 
-  private void getPermissions() {
+  private void getPermissions(String permissionType) {
```

Comment:
I would refactor permissionType to perhaps accountType seeing as we're changing the query based on the type of the account, not necessarily permissions

Reply:
resolved